src/clients/ZILClient.ts

```diff
@@ -365,6 +365,19 @@ export class ZILClient {
     return addressHex
   }
 
+  public async retrieveZRC2Info(address: string) {
+    const zilliqa = new Zilliqa(this.getProviderUrl())
+    const smartContractInit = await zilliqa.blockchain.getSmartContractInit(address)
+    const mask = ['name', '_this_address', 'symbol', 'decimals']
+    const result = smartContractInit.result
+    if (result === undefined) throw new Error(`address ${address} is invalid`)
```

Comment:
On hindsight, maybe it would be better to change the error msg to `token contract ${address} is not found`